ballerina/vector_store.bal

```diff
@@ -0,0 +1,245 @@
+// Copyright (c) 2025 WSO2 LLC (http://www.wso2.com).
+//
+// WSO2 LLC. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/ai;
+import ballerina/log;
+import ballerina/sql;
+import ballerina/uuid;
+import ballerinax/postgresql;
+import ballerinax/postgresql.driver as _;
+
+# Pgvector Vector Store implementation with support for Dense, Sparse, and Hybrid vector search modes.
+#
+# This class implements the ai:VectorStore interface and integrates with the Pgvector vector database
+# to provide functionality for vector upsert, query, and deletion.
+#
+public isolated class VectorStore {
+    *ai:VectorStore;
+    final postgresql:Client dbClient;
+    private final int vectorDimension;
+    private string tableName = "vector_store";
+    private final ai:VectorStoreQueryMode embeddingType;
+
+    # Initializes the pgvector vector store with the provided configuration.
+    #
+    # + configs - Contains configuration for database connections and other necessary parameters
+    public isolated function init(Configuration configs) returns error? {
```

Comment:
Shouldn't we add the similarity metric as a configuration here? For example: Euclidean, Manhattan, Cosine, etc.
Whats the default metric we use?